File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/Proxy/web/AbstractProxyExtractor.java
package ru.kmorozov.gbd.core.logic.Proxy.web;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import ru.kmorozov.gbd.utils.HttpConnections;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by km on 17.12.2016.
 */
public abstract class AbstractProxyExtractor {

    public List<String> getProxyList() {
        try {
            Document doc = Jsoup.connect(this.getProxyListUrl()).userAgent(HttpConnections.USER_AGENT).get();
            return this.extractProxyList(doc);
        } catch (IOException e) {
        }

        return new ArrayList<>();
    }

    protected abstract String getProxyListUrl();

    protected abstract List<String> extractProxyList(Document doc);
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/Proxy/web/AbstractProxyExtractor.kt
package ru.kmorozov.gbd.core.logic.Proxy.web

import org.jsoup.Jsoup
import org.jsoup.nodes.Document
import ru.kmorozov.gbd.utils.HttpConnections

import java.io.IOException
import java.util.ArrayList

/**
 * Created by km on 17.12.2016.
 */
abstract class AbstractProxyExtractor {

    val proxyList: MutableList<String>
        get() {
            try {
                val doc = Jsoup.connect(proxyListUrl).userAgent(HttpConnections.USER_AGENT).get()
                return extractProxyList(doc)
            } catch (e: IOException) {
            }

            return ArrayList()
        }

    protected abstract val proxyListUrl: String

    protected abstract fun extractProxyList(doc: Document): MutableList<String>
}
